core-common/src/main/java/org/mapstruct/BuilderMapping.java

```diff
@@ -0,0 +1,38 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * @author Filip Hrisafov
+ * @since 1.3
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target({
+    ElementType.TYPE,
+    ElementType.METHOD
+})
+public @interface BuilderMapping {
```

Comment:
You are right. It might not make sense to `MapMapping`. However, it will make sense during the computation of the intermediate methods for the keys / values mappings. In my opinion we need to change the `MapperConfig` and allow use of meta annotations in 2.0. The mapper config might even not be needed.

Reply:
Yes we do that in 2.0. I will apply the changes here.